src/SplitTrilinearInterpolation.jl

```diff
@@ -0,0 +1,174 @@
+module SplitTrilinearInterpolation
+using DocStringExtensions
+
+import ..Interpolation1D: Linear, Flat, interpolate1d!
+import ..BilinearInterpolation: Bilinear, interpolatebilinear!, get_dims
+
+"""
+    SplitTrilinearHV{VS, VT, HB, W1, W2}
+
+This struct stores the source and target grids and work arrays for split trilinear interpolation.
+
+# Fields
+$(DocStringExtensions.FIELDS)
+"""
+struct SplitTrilinearHV{VS, VT, HB, W1, W2}
+    "Vertical source grid"
+    vsource::VS
+    "Vertical target grid"
+    vtarget::VT
+    "bilinear object containing the horizontal source and target grids"
+    bilinear::HB
+    "work array, storing fsource on the target horizontal grid"
+    fsourceintermediate::W1
+    "vertical source grid corresponding to target horizontal grid"
+    vsourceintermediate::W2
```

Comment:
```suggestion
    "work array, storing fsource on the target horizontal grid"
    fsource_intermediate::W1
    "vertical source grid corresponding to target horizontal grid"
    vsource_intermediate::W2
```
I find using the snake case naming convention is more readable